File: Week4Java/app/src/main/java/unram/psti/week4/Sub2Activity.java
package unram.psti.week4;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.widget.TextView;

public class Sub2Activity extends AppCompatActivity {

    public static String EXTRA_DATA = "data";


    private String receivedData = null;
    private TextView txtData;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_sub2);

        txtData = findViewById(R.id.txt_data);


        Mahasiswa receivedData = getIntent().getParcelableExtra(EXTRA_DATA);

        receivedData.setNama("Ilham");

//        String mahasiwa = "nama : " + receivedData.getNama() + "\t" + "nim: " + receivedData.getNim();

        Mahasiswa receivedData2 = getIntent().getParcelableExtra(EXTRA_DATA);

        String mahasiwa2 = "nama : " + receivedData2.getNama() + "\t" + "nim: " + receivedData2.getNim();

        txtData.setText(mahasiwa2);


    }
}

File: Week4Kotlin/app/src/main/java/unram/psti/week4kotlin/Sub2Activity.kt
package unram.psti.week4kotlin

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.TextView

class Sub2Activity : AppCompatActivity() {
    companion object {
        const val EXTRA_NAME = "EXTRA_NAME"
    }
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_sub2)

        val txtData: TextView = findViewById(R.id.txt_data)

        val data = intent.getStringExtra(EXTRA_NAME)
        val str = "Data yang dikirim: $data"
        txtData.text = str

    }
}